Fix support for old rubies

ivoanjo opened the pull request:
Hello there! 👋 

While adding `ruby2_keywords` to @datadog's [`ddtrace`](https://github.com/DataDog/dd-trace-rb) gem I discovered that it had a few incompatibilities with older rubies.

I went through every one of them, and replaced them with alternatives that work as far back as Ruby 2.0.
Out of curiosity, I tested 1.9 as well, and found there were further incompatibilities but I decided to stop at that point 🤣 .

Hopefully this is acceptable :)

ivoanjo commented:
Apologies for the extra noise! I hadn't spotted the existing GitHub CI setup when I first opened the PR so I updated it as well now.

Ready for review, and thanks a lot! :)

nobu commented:
Cherry-picked some commits.
Could you rebase the rest?

And I’m not sure if docker-compose.yml is useful only on GitHub.
Do you intend to checked it out and use locally?

ivoanjo commented:
Thanks @nobu! I've rebased the two last commits.

> And I’m not sure if docker-compose.yml is useful only on GitHub.
> Do you intend to checked it out and use locally?

Yes, the `docker-compose.yml` is only for local testing. I added it as it was helpful for my own local testing (it's kinda hard to get these old rubies installed nowadays), but if you don't think it's useful, I can remove it.

Also, it would be great to have a 0.0.5 release with these fixes :) :) :)
